File: cakk-api/src/main/java/com/cakk/api/service/like/LikeService.java
package com.cakk.api.service.like;

import org.springframework.stereotype.Service;

import lombok.RequiredArgsConstructor;

import com.cakk.api.annotation.DistributedLock;
import com.cakk.core.facade.cake.CakeShopReadFacade;
import com.cakk.core.facade.user.UserLikeFacade;
import com.cakk.domain.mysql.entity.shop.CakeShop;
import com.cakk.domain.mysql.entity.user.User;

@Service
@RequiredArgsConstructor
public class LikeService {

	private final CakeShopReadFacade cakeShopReadFacade;
	private final UserLikeFacade userCakeFacade;

	@DistributedLock(key = "#cakeShopId")
	public void likeCakeShop(final User user, final Long cakeShopId) {
		final CakeShop cakeShop = cakeShopReadFacade.findByIdWithLike(cakeShopId);

		userCakeFacade.likeCakeShop(user, cakeShop);
	}
}

File: cakk-core/src/main/kotlin/com/cakk/core/service/like/LikeService.kt
package com.cakk.core.service.like

import org.springframework.stereotype.Service

import com.cakk.core.annotation.DistributedLock
import com.cakk.core.facade.cake.CakeShopReadFacade
import com.cakk.core.facade.user.UserLikeFacade
import com.cakk.infrastructure.persistence.entity.user.UserEntity

@Service
class LikeService(
	private val cakeShopReadFacade: CakeShopReadFacade,
	private val userLikeFacade: UserLikeFacade
) {

	@DistributedLock(key = "#cakeShopId")
	fun likeCakeShop(userEntity: UserEntity, cakeShopId: Long) {
		val cakeShop = cakeShopReadFacade.findByIdWithLike(cakeShopId)

		userLikeFacade.likeCakeShop(userEntity, cakeShop)
	}
}
